src/librustc_session/options.rs

```diff
@@ -849,6 +880,10 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "exclude the pass number when dumping MIR (used in tests) (default: no)"),
     dump_mir_graphviz: bool = (false, parse_bool, [UNTRACKED],
         "in addition to `.mir` files, create graphviz `.dot` files (default: no)"),
+    dump_mir_spanview: Option<MirSpanview> = (None, parse_mir_spanview, [UNTRACKED],
```

Comment:
@tmandry I checked in additional changes as a separate commit in the same PR.

The latest change was to extract the HTML+CSS doc generation into a separate crate. This cleans up the InstrumentCoverage pass a bit more; but it also makes the feature available to extend the `mir_dump`.

Now in addition to -Zdump_mir_graphviz, you can also or optionally add `-Zdump_mir_spanview`. There are three options with this flag: block (what InstrumentCoverage uses, in essence), terminator, and statement.